### ISSUE TITLE
Media Manager: asset rotation selection button creates unpredictable result

### ISSUE THREAD

**Issue by mbeganyi-a11y** [joomla/joomla-cms#48427]
### Steps to reproduce the issue

**NOTE**: Accessibility backlog issue and work-in-progress. This issue has been opened to account for all the issues flagged in the initial [Milestone 1a Audit Report](https://accessibility-project.joomla.org/results/Milestone-1a/milestone1/deliverable1/wcag-report). If this is a duplicate issue, it will be closed by the end of September 2026.

Milestone 1a   finding ID | M1a-011
-- | --
Backlog   priority | High
Severity —   carried from Milestone 1a | Serious
Implementation   effort | Medium
Target   milestone(s) | MS2a, MS2b,   MS3a, MS3b, MS5a, MS7b, MS8a, MS10
WCAG 2.2   success criterion | 3.2.1 On   Focus
Responsibility   tag(s) from report | Design,   Development
Affected   Joomla area/component | https://accessibility-project.joomla.org/results/Milestone-1a/milestone1/deliverable1/wcag-report#62379b3f-c3b6-4fa7-86a1-1d23413bebeb
Affected user   journey(s) | 06 – Media   Manager (com_media)
Source   report/reference | Milestone 1a   WCAG audit report
STF reporting   reference | To be   assigned
Issue owner | To be   assigned
Status | Triage

<p class=MsoNormal>Audit finding<o:p></o:p></p>

<p class=MsoNormal>Rotation selection button create unpredictable result<o:p></o:p></p>

<p class=MsoNormal>User impact<o:p></o:p></p>

<p class=MsoNormal>The issue creates an accessibility barrier in the component.<o:p></o:p></p>

<p class=MsoNormal>Remediation plan<o:p></o:p></p>

<p class=MsoNormal>Refactor the interaction so it uses correct semantics and states and behaves predictably.<o:p></o:p></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'>Steps to reproduce</b><o:p></o:p></p>

<p class=MsoNormal>1. Open Media Manager and select an image.<o:p></o:p></p>

<p class=MsoNormal>2. Activate the rotation control with the pointer.<o:p></o:p></p>

<p class=MsoNormal>3. Repeat using the keyboard.<o:p></o:p></p>

<p class=MsoNormal>4. Inspect the resulting image presentation.<o:p></o:p></p>

<p class=MsoNormal>5. Observe an unpredictable result or insufficient feedback.<o:p></o:p></p>

</body>

</html>

### Version

6.1

### Expected result

Rotation of an asset would occur in a predictable and logical way based on initial asset state. Suggested fix: Add an apply button and let users select an angle first and then apply it. Alternatively make the buttons buttons instead of radio buttons, so that users have the ability to skip options they don’t want to apply. Also make clear that the buttons add the degrees to the current rotation. Consider displaying somewhere what the combined rotation is.

### Actual result

When using the Angle Buttons when editing a graphic or photo, when using the keyboard, the rotation is automatically applied using the arrow keys to interact with the (invisible radio buttons. This results in the keyboard focusing the 0 degree button, when using the arrow key to go to 90, the image automatically flips 90 degrees. Going to the next button does not flip the image 180 degrees from the initial setting, but 180 degrees from the now 90 degree image. Going to 270 degrees then rotates an additional 270 degrees.

### What device were you using? (eg, phone, laptop)

Desktop web

### What web browser were you using? (eg, Chrome, Safari)

Chrome, Firefox, Edge, Safari

### Assistive Technology?

N/A

**Comment by brianteeman** [comment]
related issue https://github.com/joomla/joomla-cms/issues/29943

**Comment by mbeganyi-a11y** [comment]
Yeah I see that's been around for a while. Thanks for linking that here.
